common/flatpak-dir.c

```diff
@@ -1853,69 +1867,181 @@ repo_pull_one_dir (OstreeRepo          *self,
                    GCancellable        *cancellable,
                    GError             **error)
 {
-  GVariantBuilder builder;
   gboolean force_disable_deltas = (flatpak_flags & FLATPAK_PULL_FLAGS_NO_STATIC_DELTAS) != 0;
   g_autofree char *remote_and_branch = NULL;
   g_autofree char *current_checksum = NULL;
-  g_autoptr(GVariant) options = NULL;
   g_autoptr(GVariant) old_commit = NULL;
   g_autoptr(GVariant) new_commit = NULL;
-  const char *refs_to_fetch[2];
   const char *revs_to_fetch[2];
+  gboolean res = FALSE;
   guint32 update_freq = 0;
+#ifdef FLATPAK_ENABLE_P2P
+  g_autofree gchar *collection_id = NULL;
+#endif  /* FLATPAK_ENABLE_P2P */
 
   /* We always want this on for every type of pull */
   flags |= OSTREE_REPO_PULL_FLAGS_BAREUSERONLY_FILES;
 
-  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{sv}"));
+#ifdef FLATPAK_ENABLE_P2P
+  if (!ostree_repo_get_remote_option (self, remote_name, "collection-id", NULL,
+                                      &collection_id, NULL))
+    g_clear_pointer (&collection_id, g_free);
 
-  if (dirs_to_pull)
+  if (collection_id != NULL)
```

Comment:
Maybe there is some subsequent refactoring to do here. I left it as-is because it was easier, and also makes the review a bit easier than a more complex refactor.

Something to address once this branch is merged?